TST: add some tests of np.log for complex input.

WarrenWeckesser opened the pull request:
These are values where some platforms might lose precision.

[skip circle]

WarrenWeckesser committed:
TST: add some tests of np.log for complex input.

These are values where some platforms might lose precision.

[skip circle]

mattip commented:
Thanks. The failing tests are helpful (points out where we should be blocklisting clog on musllinux, cygwin, and pyodide). I wonder if at some point we should adopt the [cpython cmath tests](https://github.com/python/cpython/blob/main/Lib/test/test_cmath.py) with the extensive [test cases](https://github.com/python/cpython/blob/main/Lib/test/cmath_testcases.txt) for at least `cdouble`.

WarrenWeckesser committed:
BLD: blocklist complex log function clog on musl.

[skip circle]

WarrenWeckesser commented:
Reviving this PR; I rebased and force pushed the original commit.

Following the example of https://github.com/numpy/numpy/pull/25092, where the complex trig functions were blocklisted for musl, I added the appropriate `undef`s in `npy_config.h` for the complex log functions.

The pyodide tests also pass now, which I didn't expect, but I don't know how pyodide's libc affects the macros that are used in `npy_config.h`.

WarrenWeckesser committed:
BLD: blocklist complex log functions clog* on cygwin.

[skip circle]

WarrenWeckesser commented:
Also block the use of the provided complex log functions on cygwin.  cygwin uses newlib, and newlib's implementation of the complex log is the same naive implementation as that of musl.

mattip commented:
Thanks @WarrenWeckesser
